docs/spec/enums.rst

```diff
@@ -0,0 +1,333 @@
+the Enum behaviors that should be supported by type checkers and others which
+may be supported optionally. It is recommended that library and type stub
+authors avoid using optional behaviors because these may not be supported
+by some type checkers.
+
+
+Enum Definition
+---------------
+
+Enum classes can be defined using a "class syntax" or a "function syntax".
+The function syntax offers several ways to specify enum members: names passed
+as individual arguments, a list or tuple of names, a string of
+comma-delimited or space-delimited names, a list or tuple of tuples that contain
+name/value pairs, and a dictionary of name/value items.
+
+Type checkers should support the class syntax, but the function syntax (in
+its various forms) is optional::
+
+    class Color1(Enum): # Supported
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+
+    Color2 = Enum('Color2', 'RED', 'GREEN', 'BLUE')  # Optional
+    Color3 = Enum('Color3', ['RED', 'GREEN', 'BLUE'])  # Optional
+    Color4 = Enum('Color4', ('RED', 'GREEN', 'BLUE'))  # Optional
+    Color5 = Enum('Color5', 'RED, GREEN, BLUE')  # Optional
+    Color6 = Enum('Color6', 'RED GREEN BLUE')  # Optional
+    Color7 = Enum('Color7', [('RED': 1), ('GREEN': 2), ('BLUE': 3)])  # Optional
+    Color8 = Enum('Color8', (('RED': 1), ('GREEN': 2), ('BLUE': 3)))  # Optional
+    Color9 = Enum('Color9', {'RED': 1, 'GREEN': 2, 'BLUE': 3})  # Optional
+
+If a type checker supports the functional syntax, it should enforce name
+consistency. That is, if the type is assigned to a variable, the name of
+the variable must match the name of the enum class::
+
+  WrongName = Enum('Color', 'RED GREEN BLUE')  # Type checker error
+
+Enum classes can also be defined using a subclass of ``enum.Enum`` or any class
+that uses ``enum.EnumType`` (or a subclass thereof) as a metaclass. Note that
+``enum.EnumType`` was named ``enum.EnumMeta`` prior to Python 3.11. Type
+checkers should treat such classes as enums::
+
+    class CustomEnum1(Enum):
+        pass
+    
+    class Color7(CustomEnum1):  # Supported
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+
+    class CustomEnumType(EnumType):
+        pass
+    
+    class CustomEnum2(metaclass=CustomEnumType):
+        pass
+
+    class Color8(CustomEnum2):  # Supported
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+
+
+Enum Behaviors
+--------------
+
+Enum classes are iterable and indexable, and they can be called with a value
+to look up the enum member with that value. Type checkers should support these
+behaviors::
+
+    class Color(Enum):
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+
+    for color in Color:
+        reveal_type(color)  # Revealed type is 'Color'
+
+    reveal_type(Color["RED"])  # Revealed type is 'Literal[Color.RED]' (or 'Color')
+    reveal_type(Color(3))  # Revealed type is 'Literal[Color.BLUE]' (or 'Color')
+
+Calling an enum class does not invoke its constructor, as is normal
```

Comment:
I find the wording of this paragraph a little confusing, especially the second sentence. After reading it a few times, I think the second sentence is describing enum's runtime behavior? But at first I thought it was directing type checkers on how to implement `enum.__new__`. What do you think of rewording the paragraph like so, to focus more on the effect of calling an enum class and less on what's happening under the hood?

> Unlike most Python classes, calling an enum class does not invoke its constructor. Instead, the call performs a value-based lookup of an enum member.

Reply:
Thanks, I like your proposed wording better than my original — definitely clearer.